iteration.go

```diff
@@ -233,6 +262,39 @@ func createIterationLinks(request *goa.RequestData, id interface{}) *app.Generic
 	}
 }
 
+// ConvertFromLtreeFormat converts data to UUID format from ltree format.
+func ConvertFromLtreeFormat(uuid string) string {
+	// Ltree allows only "_" as a special character.
+	return strings.Replace(uuid, "_", "-", -1)
+}
+
+// IterationIDMap contains a map that will hold iteration's ID as its key
+type IterationIDMap map[uuid.UUID]*iteration.Iteration
+
+func parentPathResolver(itrMap IterationIDMap) IterationConvertFunc {
+	return func(request *goa.RequestData, itr *iteration.Iteration, appIteration *app.Iteration) {
+		parentUUIDStrings := strings.Split(iteration.ConvertFromLtreeFormat(itr.ParentPath), iteration.PathSepInService)
+		parentUUIDs := convertToUUID(parentUUIDStrings)
+		pathResolved := ""
+		for _, id := range parentUUIDs {
+			if i, ok := itrMap[id]; ok {
+				pathResolved += iteration.PathSepInService + i.Name
+			}
+		}
+		appIteration.Attributes.ResolvedParentPath = &pathResolved
+	}
+}
+
+func convertToUUID(uuidStrings []string) []uuid.UUID {
```

Comment:
we could move this function to the package resource and there to have such a function. It'll be then used by iteration.go and area.go.